scripts/release.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+set -e
+
+REPOSITORY="${REPOSITORY:-git@github.com:etcd-io/etcd-operator.git}"
+REGISTRY="${REGISTRY:-gcr.io/etcd-development/etcd-operator}"
+CONTAINER_TOOL="${CONTAINER_TOOL:-docker}"
+
+function main {
+  if [ -z "$1" ]; then
+    echo "Usage: $0 <release version>"
+    exit
+  fi
+
+  local version="v${1#v}"
+  local image="${REGISTRY}:${version}"
+
+  build_and_commit_dist_files "${image}"
+
+  if ! tag_repository "$version"; then
+    echo "Failed to tag repository"
+    exit 1
+  fi
+
+  build_and_push_image "${image}"
+}
+
+function build_and_push_image {
+  if "${CONTAINER_TOOL}" manifest inspect "${image}" &>/dev/null; then
+    echo "Skipping pushing container image. Image ${image} already exists."
+    return
+  fi
+  echo "Building and pushing container image ${image}"
+  make docker-buildx IMG="${image}"
+}
+
+function build_and_commit_dist_files {
+  local image="$1"
+
+  echo "Building distribution YAML and committing to repository"
+  make build-installer IMG="${image}"
+  git checkout -- config/manager/kustomization.yaml
+  git add dist/install.yaml
```

Comment:
Previously dist isn't managed by git. If we want to manage it in git, can we add the version into the file name, i.e. install_v0.1.0.yaml?

Reply:
Done. I named it with a hyphen for consistency with branch names, and other places where we add the version to the binary (i.e., install-v0.1.0.yaml).